src/AWSCredentials.jl

```diff
@@ -461,15 +461,18 @@ function credentials_from_webtoken()
 
     has_all_keys = 
         haskey(ENV, token_role_arn) &&
-        haskey(ENV, token_role_session) &&
         haskey(ENV, token_web_identity)
 
     if !has_all_keys
         return nothing
     end
 
     role_arn = ENV[token_role_arn]
-    role_session = ENV[token_role_session]
+    role_session = get(ENV, token_role_session) do
+        default_role_session = replace("default-AWS.jl-role-session-$(now())", ":" => "-")
```

Comment:
You can just specify a date format to avoid having to replace characters. I'd recommend sticking to a ISO8601 format so I switched to using a UTC format which doesn't use special characters:
```suggestion
        timestamp = Dates.format(now(UTC), dateformat"yyyymmdd\THHMMSSZ")
        default_role_session = "default-AWS.jl-role-session-$timestamp" 
```
Note: this module will need to have `using Dates` for this to work